File: src/main/java/com/example/probation/core/entity/VerificationToken.java
package com.example.probation.core.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.Calendar;
import java.util.Date;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "verification_token")
public class VerificationToken extends AbstractEntity {
    private static final int EXPIRATION = 60 * 2;

    private String token;
    @OneToOne(targetEntity = User.class, fetch = FetchType.EAGER)
    @JoinTable(name = "user_token", joinColumns = @JoinColumn(name = "token_id", referencedColumnName = "id"), inverseJoinColumns = @JoinColumn(name = "user_id", referencedColumnName = "id"))
    private User user;
    private Date expiryDate;

    public VerificationToken(final String token, final User user) {
        super();
        this.token = token;
        this.user = user;
        this.expiryDate = calculateExpiryDate();
    }

    private Date calculateExpiryDate() {
        var cal = Calendar.getInstance();
        cal.setTime(new Timestamp(cal.getTime().getTime()));
        cal.add(Calendar.MINUTE, VerificationToken.EXPIRATION);

        return new Date(cal.getTime().getTime());
    }
}

File: src/main/kotlin/com/example/probation/core/entity/VerificationToken.kt
package com.example.probation.core.entity

import java.time.LocalDateTime
import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.JoinColumn
import javax.persistence.JoinTable
import javax.persistence.OneToOne
import javax.persistence.Table

@Entity
@Table(name = "verification_token")
class VerificationToken(
    @Column(name = "token")
    var token: String? = null,
    @OneToOne(
        targetEntity = User::class, fetch = FetchType.EAGER,
        cascade = [CascadeType.PERSIST, CascadeType.REFRESH, CascadeType.MERGE, CascadeType.DETACH]
    )
    @JoinTable(
        name = "user_token",
        joinColumns = [JoinColumn(name = "token_id", referencedColumnName = "id")],
        inverseJoinColumns = [JoinColumn(name = "user_id", referencedColumnName = "id")]
    )
    var user: User? = null,
    @Column(name = "expiryDate")
    var expiryDate: LocalDateTime? = null
) : AbstractEntity() {
    companion object {
        const val EXPIRATION: Long = 60 * 2
    }

    constructor(_token: String, _user: User) : this() {
        token = _token
        user = _user
        expiryDate = calculateExpiryDate()
    }

    private fun calculateExpiryDate(): LocalDateTime =
        LocalDateTime.now().plusMinutes(EXPIRATION)
}
